pymeasure/instruments/anritsu/anritsuMS2090A.py

```diff
@@ -0,0 +1,516 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    truncated_range,
+    double_validation_value_and_freq
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class AnritsuMS2090A(Instrument):
+    """Anritsu MS2090A Handheld Spectrum Analyzer."""
+
+    def __init__(self, adapter, **kwargs):
+        """Constructor."""
+        self.analysis_mode = None
+        super().__init__(
+            adapter, name="Anritsu MS2090A Handheld Spectrum Analyzer", **kwargs)
+
+    #############
+    #  Mappings #
+    #############
+
+    ONOFF = ["ON", "OFF"]
+    ONOFF_MAPPING = {True: 'ON', False: 'OFF', 1: 'ON', 0: 'OFF', 'ON': 'ON', 'OFF': 'OFF'}
+    OFFFIRSTREPEAT = ['OFF', 'FIRSt', 'REPeat']
+    SPAMODES = ["SPECtrum", "NRADio", "RTSA", "LTE", "EMFMeter", "PANalyzer"]
+    UNITSFREQ = ['Hz', 'kHz', 'MHz', 'GHz']
+    UNITSFREQ_MAP = []
+
+    ####################################
+    #              GPS                 #
+    ####################################
+
+    gps_full = Instrument.measurement(
+        "FETCh:GPS:FULL?",
+        '''
+        Returns the timestamp, latitude, longitude, altitude, and satellite count of the device.

```

Comment:
Please look at the documentation regarding [docstrings](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings). You will find examples of good docstrings in the [adding instruments documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html).
Especially, The first sentence of a docstring (Python stlye) should be an imperative. Further, the docstrings of `measurement`, `setting`, and `control` should start with "Measure", "Set", "Control" (pymeasure style) as you cannot see in the documentation, whether a property is only gettable, only settable, or both.